src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -38,10 +40,16 @@ namespace AppInstaller::CLI::Workflow
 
         std::shared_ptr<ISource> OpenNamedSource(Execution::Context& context, std::string_view sourceName)
         {
-            std::shared_ptr<Repository::ISource> source;
+            std::shared_ptr<Repository::ISource> source;           
             try
             {
-                auto result = context.Reporter.ExecuteWithProgress(std::bind(Repository::OpenSource, sourceName, std::placeholders::_1), true);
+                AdditionalSourceData sourceData;
+                if(context.Contains(Execution::Data::Header))
+                {
+                    sourceData.Header = context.Get<Execution::Data::Header>();
+                }
+
+                auto result = context.Reporter.ExecuteWithProgress(std::bind(Repository::OpenSource, sourceName, std::placeholders::_1, std::move(sourceData)), true);

```

Comment:
I thought you were adding it to the SourceDetails...

Reply:
Changed source details to have CustomHeader